### PR TITLE
Support for Dev Pools

### PR DESCRIPTION
**Pull request by azlam-abdulsalam**
Fixes #540 

- Major refactor to support dev pools, including new schema
- Existing commands has to be fully deprecated with a simplified/unified schema

For eg: Pools will be created with the below schema
```
{
    "tag":"test2",
    "maxallocation":2,
    "expiry":1,
    "configFilePath":"config/project-scratch-def.json",
     "installAll":true,
     "installAsSourcePackages":true
}
```
- Validate schema on prepare command
- Validate prerequisite fields on every pool related command
- Break monolith classes such as ScratchOrgUtils to maintainable small chunks
- Use neverthrow to support better error handling
- Utilize refactored SFPLogger with better signatures
- Remove dependency on static variables to determine logging outputs

@Caitlyn-Mills  this is a significant change, hence our entire documentation need to be updated and communicated. I will create another issues ot track that.

### CODE REVIEW

**Comment by azlam-abdulsalam** [comment]
@aly76  I believe the distinction between cipool and devpool in the schema could be eliminated depending on the success of 'push' functionality, with pools using push rather than deploy

File: packages/sfpowerscripts-cli/package.json [hunk 1]
```diff
@@ -23,16 +23,16 @@
     "fs-extra": "^9.0.1",
     "glob": "^7.1.6",
     "handlebars": "^4.7.7",
-    "hook-std": "^2.0.0",
     "js-yaml": "^4.0.0",
+    "jsforce": "^1.10.1",
```
Comment (aly76) [inline packages/sfpowerscripts-cli/package.json]: Remove dependency on jsforce

File: packages/core/src/logger/SFPLogger.ts [hunk 2]
```diff
@@ -9,50 +9,61 @@ export enum LoggerLevel {
   WARN = 40,
   ERROR = 50,
   FATAL = 60,
+  SUCCESS = 70,
```
Comment (aly76) [inline packages/core/src/logger/SFPLogger.ts]: Remove LoggerLevel `Success`

File: packages/sfpowerscripts-cli/src/SfpowerscriptsCommand.ts [hunk 3]
```diff
@@ -112,6 +116,26 @@ export default abstract class SfpowerscriptsCommand extends SfdxCommand {
...
+        SFPLogger.logLevel = LoggerLevel.TRACE
+      else if(this.flags.loglevel==="debug" || this.flags.loglevel==="DEBUG")
+        SFPLogger.logLevel = LoggerLevel.DEBUG
+      else if(this.flags.loglevel==="info" || this.flags.loglevel==="INFO")
+        SFPLogger.logLevel = LoggerLevel.INFO
+      else if(this.flags.loglevel==="warn" || this.flags.loglevel==="WARN")
+        SFPLogger.logLevel = LoggerLevel.WARN
+      else if(this.flags.loglevel==="error" || this.flags.loglevel==="ERROR")
+        SFPLogger.logLevel = LoggerLevel.ERROR
+      else if(this.flags.loglevel==="FATAL" || this.flags.loglevel==="FATAL")
```
Comment (aly76) [inline packages/sfpowerscripts-cli/src/SfpowerscriptsCommand.ts]: Typo on condition, should be lowercase

**Review by aly76 — CHANGES_REQUESTED** [review]

**Review by aly76 — APPROVED** [review]

**Pull request merged**